### PR TITLE
Add Route module to the PG plugin

### PR DESCRIPTION
**Pull request by sonyy-m**

### CODE REVIEW

File: includes/rzp_route.php [hunk 1]
```diff
@@ -0,0 +1,955 @@
...
+                return $item[$column_name];
+
+            default:
+
+                return print_r($item, true); //Show the whole array for troubleshooting purposes
+        }
+    }
+
+
+    function usort_reorder($a, $b)
```
Comment (ChetanGN) [inline includes/rzp_route.php:144]: @sonyy-m Is this argument needs to be passed for this function?

File: includes/rzp_route.php [hunk 2]
```diff
@@ -0,0 +1,955 @@
...
+            'per_page' => $per_page,
+            'total_pages' => ceil($count / $per_page)
+        ));
+    }
+
+    function get_items($count)
+    {
+        $items = array();
+
+        $Wc_Razorpay_Loader = new WC_Razorpay();
```
Comment (ChetanGN) [inline includes/rzp_route.php:211]: Check the naming convention.

File: includes/rzp_route.php [hunk 3]
```diff
@@ -0,0 +1,955 @@
...
+
+        $api = $Wc_Razorpay_Loader->getRazorpayApiInstance();
+
+        try {
+            $transfers = $api->transfer->all(['count' => 100, 'expand[]' => 'recipient_settlement']);
```
Comment (ChetanGN) [inline includes/rzp_route.php:216]: @sonyy-m count should be dynamic right!

File: includes/rzp_route.php [hunk 4]
```diff
@@ -0,0 +1,955 @@
...
+                );
+            }
+        }
+        return $items;
+    }
+
+    function rzp_transfer_details()
+    {
+        if (empty(sanitize_text_field($_REQUEST['id'])) || null == (sanitize_text_field($_REQUEST['id']))) {
+            wp_die("This page consist some request parameters to view response");
```
Comment (ChetanGN) [inline includes/rzp_route.php:249]: Can we use wp notice to display this message!

File: includes/rzp_route.php [hunk 5]
```diff
@@ -0,0 +1,955 @@
...
+        } else {
+
+            $transfer_id = $_REQUEST['id'];
```
Comment (ChetanGN) [inline includes/rzp_route.php:252]: Sanitization is missing.

File: includes/rzp_route.php [hunk 6]
```diff
@@ -0,0 +1,955 @@
...
+        $context = stream_context_create(array(
+            'http' => array(
+                'header'  => "Authorization: Basic " . base64_encode($Wc_Razorpay_Loader->getSetting('key_id').":".$Wc_Razorpay_Loader->getSetting('key_secret'))
+            ),
+            "ssl"=>array(
+                "verify_peer"=>false,
+                "verify_peer_name"=>false,
+            ),
+        ));
+        $features_data = file_get_contents($url, false, $context);
```
Comment (ChetanGN) [inline includes/rzp_route.php:941]: @sonyy-m What if file_get_contents fails? Is it handled?

File: woo-razorpay.php [hunk 7]
```diff
@@ -1076,6 +1124,8 @@ public function updateOrder(& $order, $success, $errorMessage, $razorpayPaymentI
                 $order->payment_complete($razorpayPaymentId);
                 $order->add_order_note("Razorpay payment successful <br/>Razorpay Id: $razorpayPaymentId");
 
+                $this->transferFromPayment($orderId, $razorpayPaymentId); // creates transfers from payment
```
Comment (ChetanGN) [inline woo-razorpay.php:1127]: Can we pass order ref to this function instead of passing order id and fetchin the order details again?

Comment (sonyy-m) [reply woo-razorpay.php:1127]: Passing the $orderId as we need this to get the order items.

**Review by ramth05 — COMMENTED** [review]
1. Separation of route code.
2. File naming convention 
3. Account store in DB, to avoid copy paste and multiple input.
4. Adding route item list and import in product ( based on % of amount).

**Review by jampulanaveen — APPROVED** [review]
LGTM

File: woo-razorpay.php [hunk 8]
```diff
@@ -18,13 +18,16 @@
...
+require_once __DIR__.'/includes/razorpay-route.php';
+require_once __DIR__ . '/includes/razorpay-route-actions.php';
 
 use Razorpay\Api\Api;
 use Razorpay\Api\Errors;
 
 add_action('plugins_loaded', 'woocommerce_razorpay_init', 0);
 add_action('admin_post_nopriv_rzp_wc_webhook', 'razorpay_webhook_init', 10);
 
+
```
Comment (ramth05) [inline woo-razorpay.php:30]: remove this line.

**Review by ramth05 — COMMENTED** [review]
@sonyy-m , please make all function/variables name in camelCase only. If not restricted by WP.

**Pull request merged**